model-server/src/test/kotlin/org/modelix/model/server/ModelClientV2Test.kt

```diff
@@ -184,4 +185,28 @@ class ModelClientV2Test {
         assertFalse(success)
         assertFalse(containsRepository)
     }
+
+    @Test
+    fun `pulling existing versions pulls all referenced objects`() = runTest {
+        // Arrange
+        val url = "http://localhost/v2"
+        val modelClientForArrange = ModelClientV2.builder().url(url).client(client).build().also { it.init() }
+        val modelClientForAssert = ModelClientV2.builder().url(url).client(client).build().also { it.init() }
+        val repositoryId = RepositoryId("repo1")
+        val branchId = repositoryId.getBranchReference("my-branch")
+        modelClientForArrange.runWrite(branchId) { root ->
+            root.addNewChild("aChild", -1, null as IConceptReference?)
+        }
+
+        // Act
+        val versionPulled = modelClientForAssert.pullIfExists(branchId)!! as CLVersion
+
+        // Assert
+        val referencedEntriesInPulledVersion = versionPulled.data!!.getReferencedEntries()
+        for (referencedEntry in referencedEntriesInPulledVersion) {
+            // Check that the store also provides each referenced KVEntry.
+            // `getValue` would fail if this is not the case.
+            referencedEntry.getValue(versionPulled.store)
```

Comment:
Maybe we should do the same check recursively on the object returned by `getValue`.

Reply:
Thanks for the tip. Done.